web/thesauruses/haskell/lists.json

```diff
@@ -0,0 +1,386 @@
+    },
+    "unordered_immutable_list_start_number": {
+      "code": "0"
+    },
+    "access_element_in_unordered_immutable_list": {
+      "code": "list !! index"
+    },
+    "delete_unordered_immutable_list": {
+      "not-implemented": true
+    },
+    "name_of_mutable_hashed_list": {
+      "not-implemented": true
+    },
+    "create_a_mutable_hashed_list": {
+      "not-implemented": true
+    },
+    "insert_element_to_mutable_hashed_list": {
+      "not-implemented": true
+    },
+    "erase_element_from_mutable_hashed_list": {
+      "not-implemented": true
+    },
+    "delete_mutable_hashed_list": {
+      "not-implemented": true
+    },
+    "name_of_immutable_hashed_list": {
+      "code": "Set"
+    },
+    "create_a_immutable_hashed_list": {
+      "code": "Set.empty\nSet.fromList list\nSet.fromAscList ascendingList\nSet.fromDescList descendingList\nSet.fromDistinctAscList list\nSet.fromDistinctDescList list"
+    },
+    "insert_element_to_immutable_hashed_list": {
+      "code": "insert setToInsertTo elementToInsert"
+    },
+    "erase_element_from_immutable_hashed_list": {
+      "code": "delete setToDeleteFrom elementToDelete"
+    },
+    "delete_immutable_hashed_list": {
+      "not-implemented": true
+    },
+    "create_a_mutable_set": {
+      "not-implemented": true
+    },
+    "get_key_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_value_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_all_keys_from_mutable_set": {
+      "not-implemented": true
+    },
+    "get_all_values_from_mutable_set": {
+      "not-implemented": true
+    },
+    "swap_key_and_value_in_mutable_set": {
+      "not-implemented": true
+    },
+    "delete_mutable_set": {
+      "not-implemented": true
+    },
+    "create_a_immutable_set": {
+      "code": "import qualified Data.Map as Map\nMap.empty\nMap.singleton key value -- Map containing a single entry\nMap.fromSet generateValueFromKey set\nMap.fromList [(key1, value1), (key2, value2)]",
+      "comment": "choose `Data.Map.Strict` or `Data.Map.Lazy` depending on your requirements"
+    },
+    "get_key_from_immutable_set": {
+      "not-implemented": true
+    },
+    "get_value_from_immutable_set": {
+      "code": "map ! key"
+    },
+    "get_all_keys_from_immutable_set": {
+      "code": "keys map"
+    },
+    "get_all_values_from_immutable_set": {
+      "code": "elems map"
+    },
+    "swap_key_and_value_in_immutable_set": {
+      "not-implemented": true
+    },
+    "delete_immutable_set": {
+      "not-implemented": true
+    },
+    "find_element_at_position": {
+    },
+    "find_element_by_value": {
+      "code": "elemIndex elementToSearchFor list"
+    },
+    "find_minimum_element": {
+      "code": "minimum list",
+      "comment": "minimum works on all `Foldables` not just lists"
+    },
+    "find_maximum_element": {
+      "code": "maximum list",
+      "comment": "maximum works on all `Foldables` not just lists"
+    },
```

Comment:
What are these supposed to work on? Lists, Sets, Maps? Should I put all three in each of these?

Reply:
I put them here for lists, but if they work on a broader scale, that's fine. You can include that in the comments (which it looks like you did). So that's fine with me.